interface/patient_file/summary/demographics_full.php

```diff
@@ -792,7 +806,56 @@ function end_group() {
     $("select").change(function() {
         checkSkipConditions();
     });
+
+//This code deals with demographics before save action -
+<?php if($hook):?>
+
+	var f = $("form");
+
+	// Use hook to open the controller and get the new patient validation .
+	// when no params are sent this window will be closed from the zend controller.
+	var url ='<?php echo  $GLOBALS['web_root']."/interface/modules/zend_modules/public/patientvalidation?closeBeforeOpening=1";?>';
+	$("#submit").attr("type","button");
+	$("#submit").attr("name","btnSubmit");
+	$("#submit").attr("id","btnSubmit");
+	$("#btnSubmit").click(function( event ) {
+
+
+		<?php
+		// D in edit_options indicates the field is used in duplication checking.
+		// This constructs a list of the names of those fields.
+		$mflist = "";
+		$mfres = sqlStatement("SELECT field_id FROM layout_options " .
+			"WHERE form_id = 'DEM' AND uor > 0 AND field_id != '' AND " .
+			"edit_options LIKE '%D%' " .
+			"ORDER BY group_name, seq");
+		while ($mfrow = sqlFetchArray($mfres)) {
+			$field_id  = $mfrow['field_id'];
+			if (strpos($field_id, 'em_') === 0) continue;
+			if (!empty($mflist)) $mflist .= ",";
+			$mflist .= "'" . htmlentities($field_id) . "'";
```

Comment:
When you output variables to html, they need to be html escaped (to prevent cross scripting attacks which is injection of javascript and html within the variables). This is done using htmlspecialchars php function. htmlentities also does this but the encoding needs to be specified (thus won't support both latin and utf8 at same time). To make things easier there are shortcut function in openemr for this which are text() and attr(). See here for more details:
http://www.open-emr.org/wiki/index.php/Codebase_Security#SQL-Injection_and_Cross-Scripting_Prevention
(see step 4)

So, in this case, should be:
``````
$mflist .= "'" . text($field_id) . "'";
``````
Note we usually do this when we echo the variable, but since you are adding quotes, it is better to escape them here so you don't escape the quotes.